clientd/src/lib.rs

```diff
@@ -12,18 +14,55 @@ use tokio::sync::{broadcast, mpsc, oneshot};
 mod rpc;
 pub mod server;
 
-// TODO: remove pub and use this type to enforce invariants
-#[derive(Debug, Serialize)]
-pub struct CallbackResponse(pub Result<String, RpcError>);
+#[derive(Clone)]
+pub struct EventSubscribers(HashMap<EventKey, broadcast::Sender<Event>>);
 
-// TODO: figure out best approach for error handling
-#[derive(Debug, Serialize)]
-pub enum RpcError {
-    ClientError,
+impl EventSubscribers {
+    pub fn new() -> Self {
+        let mut se_map = HashMap::new();
+        for event in EventKey::iter() {
+            let (tx, _rx) = broadcast::channel(128);
+            se_map.insert(event, tx);
+        }
+        Self(se_map)
+    }
+}
+
+impl Default for EventSubscribers {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl AsRef<HashMap<EventKey, broadcast::Sender<Event>>> for EventSubscribers {
+    fn as_ref(&self) -> &HashMap<EventKey, broadcast::Sender<Event>> {
+        &self.0
+    }
+}
+
+pub struct FedimintClient(Client<UserClientConfig>);
+
+impl FedimintClient {
+    pub async fn new(cfg: PathBuf) -> Self {
```

Comment:
If a constructor is doing IO etc. I would name something more informative than `new`. `load_from_file` or something.

Reply:
`construct_from_file`, `new_from_file` ?

Reply:
I would go with some action-verb. `read_from`, `open_from`.